packages/elements/src/tree/helpers/renderer.ts

```diff
@@ -36,6 +37,19 @@ export class TreeRenderer extends Renderer {
       el.disabled = composer.getItemPropertyValue(item, 'disabled') === true;
       el.readonly = composer.getItemPropertyValue(item, 'readonly') === true;
       el.highlighted = composer.getItemPropertyValue(item, 'highlighted') === true;
+
+      el.setAttribute('aria-level', String(el.depth + 1));
+      el.setAttribute('aria-setsize', String(composer.getItemSiblings(item).length + 1));
+
+      if (el.parent) {
+        el.setAttribute('aria-expanded', el.expanded ? 'true' : 'false');
+      }
+
+      // In single mode, only children nodes are selectable
+      if (el.parent && !el.multiple) {
+        el.removeAttribute('aria-selected');
+      }
```

Comment:
The element should set the aria tags, not the renderer. The renderer should just set the correct state of the tree item.

Reply:
It's a bit confusing that some are written the renderer and some are in the item.

Reply:
Not sure what to do about `aria-setsize` however....

Reply:
I've looked into moving `aria-setsize` logics into `tree-item` and no luck so far. `tree-item` has no access to composer and is therefore unaware of adjacent nodes. This attribute is crucial for screen reader counting order. What do you think if go with this approach for now ?

Reply:
We'd probably need to add a property to the list-tem for setsize and just set the property in the renderer. It's a bit tricky because all of our foundational elements now handle ARIA automatically, so the tree-item should follow, as it's part or the suite.

This gets even more interesting when people want to create their own fully custom renderer, because they will likely break the accessibility and have to implement the ARIA attributes themselves.

Reply:
Good news. I just re-tested again on Windows, item order and set size seems to be automatically computed in both Narrator and NVDA. As long as items are in the DOM, it is fine. And these never worked with VoiceOver on MacOS anyways. I'll remove setsize and rely on browsers to compute these properties instead.